build.sh

```diff
@@ -335,12 +335,13 @@ download_sync() {
     TMP_DIR="$TMPDIR/sync_bin"
     mkdir -p "${TMP_DIR}"
     SYNC_TMP_TAR="${TMP_DIR}/sync-${REALM_SYNC_VERSION}.tar.xz.tmp"
+    rm -rf "$SYNC_TMP_TAR" # ensure we don't have a corrupt download
```

Comment:
Looking at it more I don't think that we should make this that much more complicated. The file that is being deleted is `$SYNC_TMP_TAR`, which if it is successfully downloaded gets moved to `$SYNC_TAR`. So the only files that could be at `$SYNC_TMP_TAR` to be deleted would be a failed (or theoretically in-progress) file. There is room for improvement in this area, but `$SYNC_TMP_TAR`, and some implicit caching is nice, but it does not seem the place for that.